### ISSUE TITLE
Fields without defaults cannot follow base class fields with defaults

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#4125]
This is an error at runtime and should be reported by ty (like we do when they are part of the same class):

```py
from dataclasses import dataclass

@dataclass
class Base:
    x: int = 1

@dataclass
class Child(Base):
    y: int  # This should be an error
```

https://play.ty.dev/7465333e-a62b-4a7f-895b-44e2e2cccaca
